src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs

```diff
@@ -0,0 +1,165 @@
+        public HostEnvironment(string environmentName, string contentRootPath)
+        {
+            EnvironmentName = environmentName;
+            ContentRootPath = contentRootPath;
+        }
+
+        public string EnvironmentName { get; set; }
+        public string ApplicationName { get; set; }
+        public string ContentRootPath { get; set; }
+        public IFileProvider ContentRootFileProvider { get; set; }
+    }
+
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            string? configFile = null;
+            string? environment = null;
+            var options = new OptionSet
+            {
+                {"c|config=", c => configFile = c},
+                {"e|environment=", e => environment = e},
+            };
+
+            var remaining = options.Parse(args);
+            if (remaining.Any())
+            {
+                Fatal(1, $"argument '{remaining[0]}' not recognized.");
+            }
+
+            RequireParameter(configFile, "--config");
+            RequireParameter(environment, "--environment");
+
+            if (!File.Exists(configFile))
+            {
+                Fatal(3, $"config file '{configFile}' does not exist.");
+            }
+
+            var configFileName = Path.GetFileNameWithoutExtension(configFile);
+            var configExtension = Path.GetExtension(configFile);
+            var configDir = Path.GetDirectoryName(configFile);
+
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Path.GetDirectoryName(configFile))
+                .AddDefaultJsonConfiguration(new HostEnvironment(environment, configDir),
+                    configFileName + "{0}" + configExtension)
+                .Build();
+
+            var config = new ServiceFabricClusterConfiguration();
+            configuration.Bind(config);
+
+            config.Validate();
+
+            var services = new ServiceCollection();
+
+            services.AddLogging(builder =>
+            {
+                builder.AddConsole();
+            });
+
+            await using var provider = services.BuildServiceProvider();
+
+            var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger("ClusterCreation");
+            var creator = new ServiceFabricClusterCreator(config, logger);
+
+            var cts = new CancellationTokenSource();
+            Console.CancelKeyPress += (_, __) => cts.Cancel();
+
+            await creator.CreateClusterAsync(cts.Token);
+        }
+
+        private static void RequireParameter([NotNull] string? parameter, string name)
+        {
+            if (string.IsNullOrEmpty(parameter))
+            {
+                Fatal(2, $"the {name} parameter is required.");
+            }
+        }
+
+        [DoesNotReturn]
+        private static void Fatal(int code, string errorMessage)
+        {
+            Console.Error.WriteLine("fatal: " + errorMessage);
```

Comment:
Would it be helpful to VSTS error this up?

Reply:
yep, ill add that